driver-core/src/main/com/mongodb/internal/operation/Operations.java

```diff
@@ -217,13 +217,14 @@ private <TResult> FindOperation<TResult> createFindOperation(final MongoNamespac
     }
 
     <TResult> DistinctOperation<TResult> distinct(final String fieldName, @Nullable final Bson filter, final Class<TResult> resultClass,
-            final Collation collation, final BsonValue comment) {
+            final Collation collation, final BsonValue comment, @Nullable final Bson hint, @Nullable final String hintString) {
         return new DistinctOperation<>(assertNotNull(namespace),
                 fieldName, codecRegistry.get(resultClass))
                 .retryReads(retryReads)
                 .filter(filter == null ? null : filter.toBsonDocument(documentClass, codecRegistry))
                 .collation(collation)
-                .comment(comment);
+                .comment(comment)
+                .hint(hint != null ? toBsonDocument(hint) : (hintString != null ? new BsonString(hintString) : null));
```

Comment:
It is fine to implement via one-liner, but I think the similar way in the above `createFindOperation()` is cleaner and more readable.

Reply:
```suggestion
        DistinctOperation<TResult> operation = new DistinctOperation<>(assertNotNull(namespace),
                fieldName, codecRegistry.get(resultClass))
                .retryReads(retryReads)
                .filter(filter == null ? null : filter.toBsonDocument(documentClass, codecRegistry))
                .collation(collation)
                .comment(comment);
        if (hint != null) {
            operation.hint(toBsonDocument(hint));
        } else if (hintString != null) {
            operation.hintString(hintString);
        }
        return operation;
```

Reply:
Thank you for the suggestion, will update!